implementations/elixir/ockam/ockly/native/ockly/src/lib.rs

```diff
@@ -0,0 +1,352 @@
+            .import_static_secret(
+                Secret::new(public_key.to_vec()),
+                ockam_vault::SecretAttributes::X25519,
+            )
+            .await?;
+        identities_ref
+            .purpose_keys()
+            .purpose_keys_creation()
+            .purpose_key_builder(&identifier, Purpose::SecureChannel)
+            .with_existing_key(key_id, SecretType::X25519)
+            .build()
+            .await
+    })
+    .map_err(|_| Error::Term(Box::new(atoms::attest_error())))?;
+    let encoded = minicbor::to_vec(purpose_key.attestation())
+        .map_err(|_| Error::Term(Box::new(atoms::attestation_encode_error())))?;
+    let mut exp_binary = NewBinary::new(env, encoded.len());
+    exp_binary.copy_from_slice(&encoded);
+    Ok(exp_binary.into())
+}
+
+#[rustler::nif]
+fn verify_purpose_key_attestation(
+    identity: Binary,
+    public_key: Binary,
+    attestation: Binary,
+) -> NifResult<bool> {
+    let identities_ref = identities_ref()?;
+    let attestation: PurposeKeyAttestation = minicbor::decode(&attestation)
+        .map_err(|_| Error::Term(Box::new(atoms::attestation_decode_error())))?;
+    let k = PublicKey::new(public_key.as_slice().to_vec(), SecretType::X25519);
+    block_future(async move {
+        let identity = identities_ref
+            .identities_creation()
+            .import(None, &identity)
+            .await
+            .map_err(|_| atoms::identity_import_error())?;
+        identities_ref
+            .purpose_keys()
+            .purpose_keys_verification()
+            .verify_purpose_key_attestation(Some(identity.identifier()), &attestation)
+            .await
+            .map_err(|_| atoms::attest_error())
+            .and_then(|data| {
+                if let PurposePublicKey::SecureChannelStaticKey(x) = data.public_key {
+                    if PublicKey::from(x).eq(&k) {
+                        Ok(true)
+                    } else {
+                        Err(atoms::invalid_attestation())
+                    }
+                } else {
+                    Err(atoms::purpose_key_type_not_supported())
+                }
+            })
+    })
+    .map_err(|reason| Error::Term(Box::new(reason)))
+}
+
+#[rustler::nif]
+fn check_identity<'a>(env: Env<'a>, identity: Binary) -> NifResult<Binary<'a>> {
+    let identities_ref = identities_ref()?;
+    let imported_identity = block_future(async move {
+        identities_ref
+            .identities_creation()
+            .import(None, &identity)
+            .await
+            .map_err(|_| atoms::identity_import_error())
+    })
+    .map_err(|reason| Error::Term(Box::new(reason)))?;
+    let id = imported_identity.identifier().to_string();
+    let mut binary = NewBinary::new(env, id.len());
+    binary.copy_from_slice(id.as_bytes());
+    Ok(binary.into())
+}
+
+#[rustler::nif]
+fn issue_credential<'a>(
+    env: Env<'a>,
+    issuer_identity: Binary,
+    subject_identifier: String,
```

Comment:
ah,  i see what you mean.   We don't enforce that,  it happens to be that way because this is used for tests so the identity is generated by us.    But will fix this so it's evident what's going on